R/methSeg.R

```diff
@@ -133,27 +142,42 @@ methSeg<-function(obj, diagnostic.plot=TRUE, join.neighbours=FALSE,  ...){
     diagnostic.plot = FALSE
   }
   
+  if(estimate.params.density>0 & estimate.params.density<1 ){
+    
+    nbr.sample = floor(length(seg.res) * estimate.params.density)
+    # estimate parameters for mclust
+    args.Mclust.esti = args.Mclust
+    args.Mclust.esti[["score.gr"]]=seg.res[ sample(1:length(seg.res), nbr.sample) ]
+    args.Mclust.esti[["diagnostic.plot"]]=FALSE
+    dens_estimate=do.call("densityFind", args.Mclust.esti)
+    
+    args.Mclust[["modelName"]] = dens_estimate$modelName
+    args.Mclust[["G"]] = 1:dens_estimate$G
```

Comment:
this doesn't help, you run the model twice with less parameters in the end, but this wasn't what I had in mind to make things faster